code/shared/os/Firewall.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+
+#ifdef _WIN32
+#include <iwscapi.h>
+#include <wscapi.h>
+#include <versionhelpers.h>
+#endif
+
+namespace os
+{
+std::wstring GetActiveFirewallName(std::wstring defaultName = L"Unknown")
+{
+#ifdef _WIN32_WINNT >= 0x0602 // Need at least windows 8 to compile this
+
+	// IWSCProductList and IWscProduct are Windows 8+ Desktop
+	if (IsWindows8OrGreater())
+	{
+		CoInitializeEx(0, COINIT_APARTMENTTHREADED);
+
+		IWSCProductList* productList = nullptr;
+		if (SUCCEEDED(CoCreateInstance(__uuidof(WSCProductList), NULL, CLSCTX_INPROC_SERVER, __uuidof(IWSCProductList), reinterpret_cast<LPVOID*>(&productList))))
+		{
+			LONG productCount = 0;
+			if (SUCCEEDED(productList->Initialize(WSC_SECURITY_PROVIDER_FIREWALL)) && SUCCEEDED(productList->get_Count(&productCount)))
+			{
+				IWscProduct* product;
+				WSC_SECURITY_PRODUCT_STATE state;
+				BSTR bstrName = nullptr;
+
+				for (ULONG i = 0; i < productCount; ++i)
+				{
+					if (SUCCEEDED(productList->get_Item(i, &product)))
```

Comment:
If WRL-ified placing it here might be better, especially if `&` doesn't ReleaseAndGetAddressOf.